### DISCUSSION TITLE
Use uWebSockets as a plug and play web server like python's web server or nginx

### DISCUSSION THREAD

**Discussion by levinem** [uNetworking/uWebSockets#1618] [General]
I would like to use uWebSockets for all of my web server needs.  I have a front end where the instructions to see the release build are:

1. `$ python -m http.server 8000`
2. Navigate to localhost:8000 in your browser

Is it possible to use uWebSockets in a similar manner?

Here is a typical flutter front end's file directory structure:

```
/build/web
  assets
    AssetManifest.json
    FontManifest.json
    NOTICES
    fonts
      MaterialIcons-Regular.ttf
      <other font files>
    <image files>
    packages
      cupertino_icons
        assets
          CupertinoIcons.ttf
    shaders
      ink_sparkle.frag
  canvaskit
    canvaskit.js
    canvaskit.wasm
    profiling
      canvaskit.js
      canvaskit.wasm
  favicon.png
  flutter.js
  flutter_service_worker.js
  index.html
  main.dart.js
  manifest.json
  version.json
```

**Comment by uNetworkingAB** [comment]
Python is a lot more widespread than uWS, why complicate things? I began an alternative to that python http server as an example but never finished it. So right now, nope

**Reply by levinem** [reply]
Very true, I think, if the functionality was made either plug and play or very easy to set up, it would be very widely used.  As an aside, thanks for building such a great product!

**Comment by e3dio** [comment]
It takes a couple lines of code to create file server with uWS, then call your server with `fileServer /directory 8000`

**Reply by levinem** [reply]
That is great to hear?  If you don't mind, what would those few lines need to be?  I am new to using this project, and am always excited to learn more.  I am using a web server like so:

```
 uWS::App().get("/*", [](auto *res, auto */*req*/) {
           <web server logic here>
	    res->end("Hello world!");
	}).listen(8000, [](auto *listen_socket) {
	    if (listen_socket) {
			std::cout << "Listening on port " << 3000 << std::endl;
	    }
	}).run();
```
